### PR TITLE
Adds contest CRUD STUFF 

### LINKED ISSUE #59: Contest CRUD
Now that we have the concept of Contests, we need to create CRUD methods for them.

### PR DESCRIPTION
**Pull request by deadlybutter**
#### What's this PR do?
Adds CRUD STUFF for Contests.
- Index, show, update, delete. stuff

#### How should this be manually tested?
Pull the branch, can you view/make/edit/delete a contest?

#### Any background context you want to provide?
*CRUD STUFF is a highly technical term I learned from The Googles.

#### What are the relevant tickets?
Fixes #59

### CODE REVIEW

**Comment by deadlybutter** [comment]
I think I will go back and add the more customy drop down thing we discussed in slack for setting the duration, I just wanted to first get the CRUD implemented to unblock all the other things

**Comment by deadlybutter** [comment]
@weerd @sbsmith86 feel free to take a look when you get a chance

File: app/Http/routes.php [hunk 1]
```diff
@@ -26,6 +26,10 @@
     Route::model('competitions', 'Gladiator\Models\Competition');
     Route::resource('competitions', 'CompetitionsController');
 
+    // Competitions
+    Route::model('contest', 'Gladiator\Models\Contest');
```
Comment (sbsmith86) [inline app/Http/routes.php]: I think this should be plural to follow the other routes

File: app/Http/Controllers/ContestController.php [hunk 2]
```diff
@@ -0,0 +1,93 @@
+<?php
+
+namespace Gladiator\Http\Controllers;
+
+use Gladiator\Models\Contest;
+use Gladiator\Http\Requests\ContestRequest;
+
+class ContestController extends Controller
```
Comment (weerd) [inline app/Http/Controllers/ContestController.php]: I think all our controllers are pluralized. So pluralize this one `ContestsController` and the name of the file ;)

File: app/Http/routes.php (hunk 1 above)
Comment (sbsmith86) [inline app/Http/routes.php]: Also all of our views directories are plural as well (`/views/competitions/`, `/views/waitingrooms/`) so maybe update contests to be the same.

File: app/Http/Controllers/ContestController.php (hunk 2 above)
Comment (deadlybutter) [inline app/Http/Controllers/ContestController.php]: yeah I noticed but `ContestsController` just sounded funny to me for some reason, I can change it if you think its better that way

File: app/Http/routes.php [hunk 3]
```diff
@@ -26,6 +26,10 @@
...
+    Route::resource('contest', 'ContestController');
```
Comment (weerd) [inline app/Http/routes.php]: update to plural version here too ;)

File: app/Http/routes.php (hunk 1 above)
Comment (deadlybutter) [inline app/Http/routes.php]: yeah ill change it, as I pointed out to Diego 'ContestsController' just sounded funny to me for some reason https://github.com/DoSomething/gladiator/pull/65#discussion_r55572885

File: resources/views/contest/partials/_form_contest.blade.php [hunk 4]
```diff
@@ -0,0 +1,20 @@
+@include('layouts.errors')
+
+<div class="form-item -padded">
+    {!! Form::label('campaign_id', 'Campaign ID:', ['class' => 'field-label']) !!}
+    {!! Form::text('campaign_id', (isset($contest->campaign_id)) ? $contest->campaign_id : NULL, ['class' => 'text-field']) !!}
```
Comment (weerd) [inline resources/views/contest/partials/_form_contest.blade.php]: switch these to use the cleaner approach of:

```php
{{ $contest->campaign_id or old('campaign_id') }}
```

File: resources/views/layouts/navigation.blade.php [hunk 5]
```diff
@@ -5,6 +5,12 @@
         @if (Auth::user())
             <ul class="navigation__primary">
                 <li>
+                    <a href="{{ URL::to('contest') }}">
+                        <strong class="navigation__title">Contests</strong>
+                        <span class="navigation__subtitle">"Death match" -Aaron</span>
```
Comment (weerd) [inline resources/views/layouts/navigation.blade.php]: ha ;)

File: app/Http/Controllers/ContestController.php (hunk 2 above)
Comment (weerd) [inline app/Http/Controllers/ContestController.php]: yeah I see what you mean, some stuff ends up sounding funny, but I'd rather follow the expected standard. I'll be expecting it to be plural when I use it and get errors :confounded:

**Comment by deadlybutter** [comment]
@weerd good for re-review!

**Comment by angaither** [comment]
stylezci

File: app/Http/Controllers/ContestsController.php [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+     * Store a newly created resource in storage.
+     *
+     * @param  \Illuminate\Http\Request  $request
+     * @return \Illuminate\Http\Response
+     */
+    public function store(ContestRequest $request)
+    {
+        $contest = Contest::create($request->all());
+
+        return redirect()->action("ContestsController@show", $contest->id)->with('status', 'Contest has been saved!');
```
Comment (weerd) [inline app/Http/Controllers/ContestsController.php]: cool!

Comment (weerd) [inline app/Http/Controllers/ContestsController.php]: use single quotes tho :nitpicky:

Comment (deadlybutter) [inline app/Http/Controllers/ContestsController.php]: (diego i think you forgot to refresh :wink: )

File: resources/views/contests/partials/_form_contest.blade.php [hunk 7]
```diff
@@ -0,0 +1,20 @@
+@include('layouts.errors')
+
+<div class="form-item -padded">
+    {!! Form::label('campaign_id', 'Campaign ID:', ['class' => 'field-label']) !!}
+    {!! Form::text('campaign_id', old('campaign_id'), ['class' => 'text-field']) !!}
```
Comment (weerd) [inline resources/views/contests/partials/_form_contest.blade.php]: missing the full `or` statement in each of these...

Comment (deadlybutter) [inline resources/views/contests/partials/_form_contest.blade.php]: Oh i thought you literally meant $thing _or_ old('dfssdf')

Comment (weerd) [inline resources/views/contests/partials/_form_contest.blade.php]: oh haha nope!

**Comment by deadlybutter** [comment]
@weerd HTML updated!

File: resources/views/contests/partials/_form_contest.blade.php [hunk 8]
```diff
@@ -0,0 +1,20 @@
+@include('layouts.errors')
+
+{{ csrf_field() }}
+
+<div class="form-item -padded">
+    <label class="field-label" for="id">Campaign ID:</label>
+    <input type="number" name="campaign_id" id="campaign_id" class="text-field" placeholder="1234" value="{{ $contest->campaign_id or old('campaign_id') }}"/>
```
Comment (weerd) [inline resources/views/contests/partials/_form_contest.blade.php]: So this is working correctly now? :)

File: resources/views/contests/show.blade.php [hunk 9]
```diff
@@ -0,0 +1,36 @@
...
+                <ul class="form-actions -inline">
+                    <li>
+                        <a href="{{ route('contests.edit', $contest->id) }}" class="button">Edit</a>
+                    </li>
+                    <li>
+                        {!! Form::open(['method' => 'DELETE','route' => ['contests.destroy', $contest->id]]) !!}
+
+                            {!! Form::submit('Delete', array('class' => 'button delete')) !!}
+
+                        {!! Form::close() !!}
```
Comment (weerd) [inline resources/views/contests/show.blade.php]: Get rid of the `Form` facade here too! :)

File: resources/views/contests/partials/_form_contest.blade.php (hunk 8 above)
Comment (deadlybutter) [inline resources/views/contests/partials/_form_contest.blade.php]: Yep

File: resources/views/contests/show.blade.php (hunk 9 above)
Comment (deadlybutter) [inline resources/views/contests/show.blade.php]: ah forgot about this one

File: resources/views/contests/partials/_form_contest.blade.php (hunk 8 above)
Comment (weerd) [inline resources/views/contests/partials/_form_contest.blade.php]: Awesome :dancer:

**Comment by weerd** [comment]
I think StyleCI is complaining cause there's extra tab spacing. Delete that and should be good!

**Comment by deadlybutter** [comment]
DIEGO I DID IT YOURE A GENIUS

**Comment by weerd** [comment]
:+1:

**Pull request merged**